xds/src/main/java/io/grpc/xds/internal/sds/SdsProtocolNegotiators.java

```diff
@@ -436,9 +442,11 @@ public void updateSecret(SslContext sslContext) {
                   InternalProtocolNegotiators.serverTls(sslContext).newHandler(grpcHandler);
 
               // Delegate rest of handshake to TLS handler
-              ctx.pipeline().addAfter(ctx.name(), null, handler);
-              fireProtocolNegotiationEvent(ctx);
-              ctx.pipeline().remove(bufferReads);
+              if (!handlerRemoved) {
```

Comment:
`ctx.isRemoved()`

Reply:
Hmmm, much simpler. Thanks I'll make this change